core/src/test/java/com/radixdlt/api/core/TransactionStreamTest.java

```diff
@@ -105,29 +105,42 @@ public void test_core_api_can_submit_and_commit_transaction_after_running_all_sc
 
       test.runUntilState(allCommittedTransactionSuccess(transaction.raw()), 100);
 
-      var lastTransaction =
+      var lastLedgerTransaction =
           (UserLedgerTransaction)
-              Iterables.getLast(
-                      getStreamApi()
-                          .streamTransactionsPost(
-                              new StreamTransactionsRequest()
-                                  .network(networkLogicalName)
-                                  .transactionFormatOptions(
-                                      new TransactionFormatOptions().rawLedgerTransaction(true))
-                                  .limit(1000)
-                                  .fromStateVersion(1L))
-                          .getTransactions())
-                  .getLedgerTransaction();
+              readAllTransactionsFromStreamAndReturnLast().getLedgerTransaction();
 
       // In order for this assertion to pass, we must have downloaded all the scenario transactions
       // in the above first page of data.
       // This ensures that genesis and all scenarios must have data which can be mapped by the
       // Core API Transaction Stream
-      assertThat(lastTransaction.getNotarizedTransaction().getPayloadHex())
+      assertThat(lastLedgerTransaction.getNotarizedTransaction().getPayloadHex())
           .isEqualTo(transaction.hexPayloadBytes());
     }
   }
 
+  private CommittedTransaction readAllTransactionsFromStreamAndReturnLast() throws Exception {
+    var fromStateVersion = 1L;
+    while (true) {
```

Comment:
_(out of curiosity)_ wow, the test was hitting a 1000?

Reply:
No - there is now a scenario with purposefully very large receipts - which triggers the "max response size" limiter and multiple pages.